### ISSUE TITLE
AUI don't detect if you need the Nvidia current drivers or legacy drivers

### ISSUE THREAD

**Issue by mariojuniorjp** [helmuthdu/aui#265]
One thing I noticed in the script is that it does not determine which of the two drivers must be installed on the system, based on the graphic card installed.

I own a G210. The correct drivers for this card are the legacy drivers that have the 304xx numbering. The aui installs the wrong drivers, specifically the number 340xx. What happens is that I have to install the free drivers for later uninstall them and install the correct one.

>3. Install the appropriate driver for your card:

        For GeForce 400 series cards and newer [NVCx and newer], install the nvidia or nvidia-lts package along with nvidia-libgl.
        For GeForce 8000/9000 and 100-300 series cards [NV5x, NV8x, NV9x and NVAx] from around 2006-2010, install the nvidia-340xx or nvidia-340xx-lts package along with nvidia-340xx-libgl.
        For GeForce 6000/7000 series cards [NV4x and NV6x] from around 2004-2006, install the nvidia-304xx or nvidia-304xx-lts package along with nvidia-304xx-libgl.

        For even older cards, have a look at #Unsupported drivers.
        For the very latest GPU models, it may be required to install the nvidia-betaAUR package, since the stable drivers may not support the newly introduced features.

4. If you are on 64-bit and also need 32-bit OpenGL support, you must also install the equivalent lib32 package from the multilib repository (e.g. lib32-nvidia-libgl, lib32-nvidia-340xx-libgl or lib32-nvidia-304xx-libgl).

**Comment by luca020400** [comment]
Can you run dmesg and pastebin the output ?

**Comment by mariojuniorjp** [comment]
Here we go: http://pastebin.com/Vj9KgFAn

**Comment by luca020400** [comment]
Can you run lspci -k | grep -A 2 -E "(VGA|3D)"

**Comment by mariojuniorjp** [comment]
01:00.0 VGA compatible controller: NVIDIA Corporation GT218 [GeForce 210] (rev a2)
	Subsystem: Micro-Star International Co., Ltd. [MSI] N210 [Geforce 210] PCIe graphics adapter
	Kernel driver in use: nouveau

**Comment by luca020400** [comment]
Are you sure you need 304xx ?
NVA8 (GT218)	GeForce 8400 GS, ION 2, GeForce 205, 210, G 210M, 305M, 310(M), 405 
Quadro FX (380 LP, 380M), NVS (300, 2100M, 3100M) --> For GeForce 8000/9000 and 100-300 series cards [NV5x, NV8x, NV9x and NVAx] from around 2006-2010, install the nvidia-340xx or nvidia-340xx-lts package along with nvidia-340xx-libgl.
That's weird

Anyway I'll try to write a small scirpt for detecting the proper driver

**Comment by mariojuniorjp** [comment]
Yes, I'm sure. If I install the 340xx driver, xorg does not start.

Look:

Fatal server error:

(EE) no screen found

xinit: giving up
xinit: unable to connect to X server: connection refused
xinit: server error ﻿

or: https://lh4.googleusercontent.com/-Ip61sRufM2k/VPDsRRwsDcI/AAAAAAAAEAI/A9kFb-_yfYg/w1440-h810-no/15%2B-%2B1

**Comment by luca020400** [comment]
https://gist.github.com/luca020400/ed05440c64626f901541
Try and post the output

**Comment by mariojuniorjp** [comment]
nvidia nvidia-utils nvidia-libgl

**Comment by luca020400** [comment]
Anyway you're using nouveau driver

**Comment by mariojuniorjp** [comment]
Excuse me. I was confusing versions. The version I need is the same 340xx. The version that installs the AUI is the latest version. So is the error I described.

I removed the nouveau driver. When configuring lilo to install the Nvidia proprietary driver it selected the latest, which is the 355.11-4.

Pacotes (2) nvidia-355.11-4  nvidia-utils-355.11-1

**Comment by mariojuniorjp** [comment]
Installed version 340xx, but still in black screen. Same thing on 304xx. Nouveau works perfectly.

**Comment by mariojuniorjp** [comment]
Hey luca, look:

dmesg 2: http://pastebin.com/e46pjyXZ

lspci: http://pastebin.com/0YZpiDyX

Now with 340xx drivers.

**Comment by helmuthdu** [comment]
I tried to implement this some time ago, in an automatic way like is today, but i give up, there is too many models and i can't test any setup, either. For now, the only way (my option) to do this is to change the way of the current video install to a more manual mode.
I kind stopped the development since i had to work in 'windows environment' (wasn't planning, but it's the life).
